### PR TITLE
AI does not autofire weapons with firing force if preparing to jump

### PR DESCRIPTION
**Pull request by tehhowch**
Refs #2147, fc785a18f37f3a1ac6a39bdd84d916d67268c4ba

 - Switched Outfit * to Weapon * and updated variable names
 - Updated the includes to reflect the used classes.

With this, while ships are being told to prepare to jump, they will not fire weaponry which imparts a firing force on their (e.g. could cause them to no longer be jump-ready). This behavior properly propagates to mission NPCs that would jump with you.

### CODE REVIEW

**Comment by jafdy** [comment]
If their own engines are enough to counter the force of forward guns then they would be quite safe to fire, simply by using the engines to hold station.

**Comment by tehhowch** [comment]
Autofire is called for both turrets and forward guns. I could add a special case to handle checking the firing angle but it would be a much larger check given that it would have to handle forward and reverse thrust, and positive and negative firing forces. Additionally, only forward guns mounted at 0 X add a force directly in line with the ship's thrust - the others converge, and thus we'd half to check if the offsetting hardpoint exists, and is also set to fire, in order to counteract the X component of the firing force.

In short, it's a lot more in depth of a check compared to "will it move me when I don't want to be moved?"
